python/ray/autoscaler/autoscaler.py

```diff
@@ -432,11 +436,16 @@ def _update(self):
         self.last_update_time = now
         num_pending = self.num_launches_pending.value
         nodes = self.workers()
-        self.log_info_string(nodes)
         self.load_metrics.prune_active_ips(
             [self.provider.internal_ip(node_id) for node_id in nodes])
         target_workers = self.target_num_workers()
 
+        if len(nodes) >= target_workers:
+            if "CPU" in self.resource_requests:
+                del self.resource_requests["CPU"]
```

Comment:
why drop?

Reply:
We've satisfied the resource request.

It doesn't act as a runtime min_workers - it's a signal to make that level of resource available ephemerally, and scale down once it's no longer used. That way, consumers of the API don't need to think about ramping down their request as tasks complete.